Code/Framework/AzToolsFramework/Tests/Prefab/Benchmark/Spawnable/SpawnAllEntitiesBenchmarks.cpp

```diff
@@ -0,0 +1,123 @@
+    }
+    BENCHMARK_REGISTER_F(BM_SpawnAllEntities, SingleEntitySpawnable_SpawnCallVariable)
+        ->RangeMultiplier(10)
+        ->Range(100, 10000)
+        ->Unit(benchmark::kMillisecond)
+        ->Complexity();
+
+    BENCHMARK_DEFINE_F(BM_SpawnAllEntities, SingleSpawnCall_EntityCountVariable)(::benchmark::State& state)
+    {
+        const unsigned int entityCountInSpawnable = static_cast<unsigned int>(state.range());
+
+        CreateSpawnable(entityCountInSpawnable);
+
+        for (auto _ : state)
+        {
+            m_spawnTicket = new AzFramework::EntitySpawnTicket(m_spawnableAsset);
+            AzFramework::SpawnableEntitiesInterface::Get()->SpawnAllEntities(*m_spawnTicket);
+            m_rootSpawnableInterface->ProcessSpawnableQueue();
+
+            // Destroy the ticket so that this queues a request to delete all the entities spawned with this ticket.
+            state.PauseTiming();
+            delete m_spawnTicket;
+            m_spawnTicket = nullptr;
+
+            // This will process the request to delete all entities spawned with the ticket
+            m_rootSpawnableInterface->ProcessSpawnableQueue();
+            state.ResumeTiming();
+        }
+
+        state.SetComplexityN(entityCountInSpawnable);
+    }
+    BENCHMARK_REGISTER_F(BM_SpawnAllEntities, SingleSpawnCall_EntityCountVariable)
+        ->RangeMultiplier(10)
+        ->Range(100, 10000)
+        ->Unit(benchmark::kMillisecond)
+        ->Complexity();
+
+    BENCHMARK_DEFINE_F(BM_SpawnAllEntities, EntityCountVariable_SpawnCallCountVariable)(::benchmark::State& state)
+    {
+        const unsigned int entityCountInSpawnable = static_cast<unsigned int>(state.range(0));
+        const unsigned int spawnCallCount = static_cast<unsigned int>(state.range(1));
+
+        CreateSpawnable(entityCountInSpawnable);
+
+        for (auto _ : state)
+        {
+            m_spawnTicket = new AzFramework::EntitySpawnTicket(m_spawnableAsset);
+
+            for (unsigned int spawnCallCounter = 0; spawnCallCounter < spawnCallCount; spawnCallCounter++)
+            {
+                AzFramework::SpawnableEntitiesInterface::Get()->SpawnAllEntities(*m_spawnTicket);
+            }
+
+            m_rootSpawnableInterface->ProcessSpawnableQueue();
+
+            state.PauseTiming();
+            delete m_spawnTicket;
+            m_spawnTicket = nullptr;
+            m_rootSpawnableInterface->ProcessSpawnableQueue();
+            state.ResumeTiming();
+        }
+
+        state.SetComplexityN(entityCountInSpawnable * spawnCallCount);
+    }
+    // Provide ranges here to compare times for spawning the same number of entities by altering entityCountInSpawnable and spawnCallCount.
+    BENCHMARK_REGISTER_F(BM_SpawnAllEntities, EntityCountVariable_SpawnCallCountVariable)
+        ->Args({ 10, 100 })
+        ->Args({ 100, 10 })
+        ->Args({ 10, 1000 })
+        ->Args({ 1000, 10 })
+        ->Args({ 100, 1000 })
+        ->Args({ 1000, 100 })
```

Comment:
Could the first benchmark be combined with this one by adding
```
->Args({ 1, 10 })
->Args({ 1, 100 })
->Args({ 1, 1000 })
```
?

Reply:
Technically, the second benchmark can also be combined with this one. But that'll create a lot of results for 1 test and makes it harder imo to interpret the results. By keeping them separate, the mental model is a bit clearer imo.

1. Test 1 : What's the performance if I spawn a single spawanble many times?
2. Test 2: What's the performance if I use a single spawnable to spawn all my entities?
3. Test 3: How do I optimize spawning by varying the spawn call count and number of entities per spawnable count?

Each of these tests can grow larger in the future and more use cases will be added, like spawning with meshes etc. So, in terms of scaling the benchmarks, it makes sense to split the workload rather than use 1 test that does everying.